servers/src/main/java/tachyon/worker/block/BlockStoreLocation.java

```diff
@@ -144,10 +146,20 @@ public String toString() {
   @Override
   public boolean equals(Object object) {
     if (object instanceof BlockStoreLocation
+        && ((BlockStoreLocation) object).tierLevel() == tierLevel()
         && ((BlockStoreLocation) object).tierAlias() == tierAlias()
         && ((BlockStoreLocation) object).dir() == dir()) {
       return true;
     }
     return false;
   }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(new Object[] {
+        mTierLevel,
+        mTierAlias,
+        mDirIndex,
+    });
+  }
```

Comment:
Thanks, I've configured IntelliJ to use Eclipse formatter according to the instruction on tachyon-project.org website